packages/console/src/monaco/MonacoUtils.ts

```diff
@@ -2,70 +2,12 @@
-import 'monaco-editor/esm/vs/editor/contrib/fontZoom/fontZoom.js';
-import 'monaco-editor/esm/vs/editor/contrib/format/formatActions.js';
-import 'monaco-editor/esm/vs/editor/contrib/gotoError/gotoError.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneGotoLineQuickAccess.js';
-import 'monaco-editor/esm/vs/editor/contrib/gotoSymbol/goToCommands.js';
-import 'monaco-editor/esm/vs/editor/contrib/gotoSymbol/link/goToDefinitionAtPosition.js';
-import 'monaco-editor/esm/vs/editor/contrib/hover/hover.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/iPadShowKeyboard/iPadShowKeyboard.js';
-import 'monaco-editor/esm/vs/editor/contrib/inPlaceReplace/inPlaceReplace.js';
-import 'monaco-editor/esm/vs/editor/contrib/indentation/indentation.js';
-import 'monaco-editor/esm/vs/editor/contrib/inlineCompletions/ghostTextController.js';
-import 'monaco-editor/esm/vs/editor/contrib/inlayHints/inlayHintsController.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/inspectTokens/inspectTokens.js';
-import 'monaco-editor/esm/vs/editor/contrib/linesOperations/linesOperations.js';
-import 'monaco-editor/esm/vs/editor/contrib/linkedEditing/linkedEditing.js';
-import 'monaco-editor/esm/vs/editor/contrib/links/links.js';
-import 'monaco-editor/esm/vs/editor/contrib/multicursor/multicursor.js';
-import 'monaco-editor/esm/vs/editor/contrib/parameterHints/parameterHints.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneCommandsQuickAccess.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneHelpQuickAccess.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/quickAccess/standaloneGotoSymbolQuickAccess.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/referenceSearch/standaloneReferenceSearch.js';
-import 'monaco-editor/esm/vs/editor/contrib/rename/rename.js';
-import 'monaco-editor/esm/vs/editor/contrib/smartSelect/smartSelect.js';
-import 'monaco-editor/esm/vs/editor/contrib/snippet/snippetController2.js';
-import 'monaco-editor/esm/vs/editor/contrib/suggest/suggestController.js';
-import 'monaco-editor/esm/vs/editor/standalone/browser/toggleHighContrast/toggleHighContrast.js';
-import 'monaco-editor/esm/vs/editor/contrib/toggleTabFocusMode/toggleTabFocusMode.js';
-import 'monaco-editor/esm/vs/editor/contrib/caretOperations/transpose.js';
-import 'monaco-editor/esm/vs/editor/contrib/unusualLineTerminators/unusualLineTerminators.js';
-import 'monaco-editor/esm/vs/editor/contrib/viewportSemanticTokens/viewportSemanticTokens.js';
-import 'monaco-editor/esm/vs/editor/contrib/wordHighlighter/wordHighlighter.js';
-import 'monaco-editor/esm/vs/editor/contrib/wordOperations/wordOperations.js';
-import 'monaco-editor/esm/vs/editor/contrib/wordPartOperations/wordPartOperations.js';
-import 'monaco-editor/esm/vs/basic-languages/typescript/typescript.contribution.js';
+import * as monaco from 'monaco-editor';
```

Comment:
Sweet. Does this have any negative effect on our bundle size?

Reply:
Languages are now lazily loaded, so I don't think it should. Also the link to the "features" that I had before no longer works, since monaco-web-plugin-packager has been rolled into the main monaco-editor repo.

That being said, I tried building and then loading it up locally, the JS that gets loaded before:
![image](https://user-images.githubusercontent.com/4505624/189201662-acbc515c-0dab-4948-bdee-724c1ac47dd4.png)

After:
![image](https://user-images.githubusercontent.com/4505624/189202081-55da73e0-2622-447e-86ba-0a5ec9aed9d1.png)

So an increase of 5.6MB to 6.1MB. I'll see if I can pare that back down (I'll need to find the full features list in Monaco-editor again since the webpack link is gone).

Reply:
I think it's fine unless we have a good reason to spend time shrinking the main app bundle size. 500kB (1 time load since it should be cached) is not bad IMO for our app. Not sure it's worth the dev time to cut that down at the moment and I think the single import for monaco is a nice improvement to the code

Reply:
We should only care about bundle size for embedded versions and the js api itself. If it's the full app, it doesn't really matter.

Reply:
Embedded grid has an increase in size as well, about 0.5mb (~15% increase):

Before the update (3.4mb):
![image](https://user-images.githubusercontent.com/4505624/189359523-c0245ea5-3719-48ce-ab1a-86933651a049.png)

With this update (3.9mb):
![image](https://user-images.githubusercontent.com/4505624/189359007-181b3b65-d732-4747-ab1f-d642a6f894b0.png)

I can see about reducing the size but I do like not having to import every feature.

Reply:
@dsmmcken  Looking into it a bit further, I don't know if I can actually reduce the size much - seems like the size increase is simply with the monaco update, not that we're importing a whole bunch of new features.

For example, using `npx source-map-explorer ` to compare, here's before:
![image](https://user-images.githubusercontent.com/4505624/189362922-379c965c-2a77-4802-afb6-f263f514e1c6.png)

And after the update:
![image](https://user-images.githubusercontent.com/4505624/189362992-bce30ee4-1c47-40ca-b647-aa94ee85b294.png)

As you can see it is really spread out over all parts of the monaco package; 100kb increase in base, 50kb increase in platform, 100kb increase in contrib, 80kb to common, etc. I'd really need to dig into which areas we can pull out, as I don't see any obvious ones (perhaps the diffEditorWidget, but that was there prior).

Reply:
Okay. Something we could consider in future is pulling monaco from embed grid entirely. I think it's just the syntax highlighter for custom column editor.

Reply:
It's also for the cell overflow modal now too (colourizing and displaying the results)

Reply:
Added a ticket: https://github.com/deephaven/web-client-ui/issues/750
We could also lazy load those features and that would split up the package size a bit, so that's an option. E.g. Don't load monaco-editor until you open a cell overflow or custom column editor.